### PR TITLE
sfx exporter: wrap MetricTranslator and logger in MetricsConverter

### PR DESCRIPTION
**Pull request by pmcollins**
**Description:** This change adds a type, MetricsConverter, which wraps a
MetricTranslator and logger. MetricsConverter has the same lifecycle as
signalfxExporter, and will be able to maintain state if desired, which
may be needed to calculate deltas.

**Testing:** Unit tests pass.

**Documentation:** Docs added to exported types/functions.

### CODE REVIEW

File: exporter/signalfxexporter/translation/converter.go [hunk 1]
```diff
@@ -208,28 +222,79 @@ func metricDataToSfxDataPoints(
 	return sfxDataPoints, numDroppedTimeSeries
 }
 
-func appendAttributesToDimensions(
-	dimensions []*sfxpb.Dimension,
-	attribs map[string]string,
-) []*sfxpb.Dimension {
+// sanitizeDataPointLabels replaces all characters unsupported by SignalFx backend
```
Comment (dmitryax) [inline exporter/signalfxexporter/translation/converter.go:225]: Is it just reshuffling of the functions? can it be another commit if we really need it? it makes it hard to review

Comment (pmcollins) [reply exporter/signalfxexporter/translation/converter.go:225]: I moved the functions over to the new file one by one, but yeah, the diff that GitHub created ended up kind of garbled. I'll see if I can make this simpler.

Comment (pmcollins) [reply exporter/signalfxexporter/translation/converter.go]: Fixed.

Comment (dmitryax) [reply exporter/signalfxexporter/translation/converter.go]: Oh I didn't notice it's a new file

**Review by dmitryax — APPROVED** [review]

File: exporter/signalfxexporter/translation/converter.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019, OpenTelemetry Authors
+// Copyright 2020, OpenTelemetry Authors
```
Comment (bogdandrutu) [inline exporter/signalfxexporter/translation/converter.go:1]: remove the year
